### PR TITLE
difft: add page

### PR DESCRIPTION
**Pull request by fazlearefin**
- [x] The page(s) are in the correct platform directories: `common`, `linux`, `osx`, `windows`, `sunos`, `android`, etc.
- [x] The page(s) have at most 8 examples.
- [x] The page description(s) have links to documentation or a homepage.
- [x] The page(s) follow the [content guidelines](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#guidelines).
- [x] The PR title conforms to the recommended [templates](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#commit-message-and-pr-title).
- **Version of the command being documented (if known):**

### CODE REVIEW

File: pages/common/difft.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+- Only report if there are changes between the source code files:
+
+`difft --check-only {{path/to/original_file}} {{path/to/new_file}}`
+
+- Show changes in different display modes (default mode is `side-by-side`):
+
+`difft --display {{side-by-side|side-by-side-show-both|inline|json}} {{path/to/original_file}} {{path/to/new_file}}`
+
+- Ignore comments when diffing:
```
Comment (vitorhcl) [inline pages/common/difft.md:18]:
```suggestion
- Ignore comments when comparing:
```
This is a more natural word for people unfamiliar with the tech context.

File: pages/common/difft.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+`difft --ignore-comments {{path/to/original_file}} {{path/to/new_file}}`
+
+- Turn on syntax highlighting of source code on or off (default is `on`):
```
Comment (vitorhcl) [inline pages/common/difft.md:22]:
```suggestion
- Enable/Disable syntax highlighting of source code (default is `on`):
```

File: pages/common/difft.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+`difft --syntax-highlight {{on|off}} {{path/to/original_file}} {{path/to/new_file}}`
+
+- Do not output anything at all if there are no changes between files:
+
+`difft --skip-unchanged {{path/to/original_file}} {{path/to/new_file}}`
+
+- Print all the languages supported by difftastic, along with their extensions:
```
Comment (vitorhcl) [inline pages/common/difft.md:30]:
```suggestion
- Print all the languages supported the tool, along with their extensions:
```
We avoid mentioning the name of the tools in the pages

File: pages/common/difft.md [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+- Check files or directories for changes:
+
+`difft {{path/to/original_file_or_directory}} {{path/to/new_file_or_directory}}`
+
+- Only report if there are changes between the source code files:
+
+`difft --check-only {{path/to/original_file}} {{path/to/new_file}}`
+
+- Show changes in different display modes (default mode is `side-by-side`):
```
Comment (vitorhcl) [inline pages/common/difft.md:14]:
```suggestion
- Specify the display mode (default is `side-by-side`):
```
I removed the "word" for consistency with the page itself.

File: pages/common/difft.md [hunk 5]
```diff
@@ -0,0 +1,32 @@
+# difft
+
+> Difftastic is a structural diff that understands syntax.
+> More information: <https://difftastic.wilfred.me.uk/introduction.html>.
```
Comment (vitorhcl) [inline pages/common/difft.md:4]:
```suggestion
> See also: `delta`, `diff`.
> More information: <https://difftastic.wilfred.me.uk/introduction.html>.
```
I think there are more pages that document similar tools.

**Review by vitorhcl — COMMENTED** [review]
Nice tool, thank you for your contribution!

Remember that a diff tool can also be used for comparing the differences between two files, not only between two versions of the same file.

IMO using original/new is not ideal, it gives the idea of change, maybe source/target?

File: pages/common/difft.md [hunk 6]
```diff
@@ -0,0 +1,32 @@
+# difft
+
+> Difftastic is a structural diff that understands syntax.
```
[the remark this answers is not in the document]
Comment (CleanMachine1) [reply pages/common/difft.md:3]:
```suggestion
> Compare files or directories based on the formatting of the text.
```
```suggestion
> Compare files or directories based on programming language in use.
```

[the remark this answers is not in the document]
Comment (vitorhcl) [reply pages/common/difft.md:3]:
```suggestion
> Compare files or directories based on the syntax of the programming language.
```
I liked your fist suggestion, but IMO "formatting of the text" is a bit vague in this context, what about this?

File: pages/common/difft.md (hunk 3 above)
Comment (fazlearefin) [reply pages/common/difft.md:31]: Should be `by the tool`

**Comment by kbdharun** [comment]
> Closes #12540

@fazlearefin Trigger actions like these only work when added in the PR description/merge commit message. I have manually linked the other issue to be closed when this gets merged.

File: pages/common/difft.md [hunk 7]
```diff
@@ -0,0 +1,33 @@
+# difft
+
+> Compare files or directories based on the syntax of the programming language.
+> See also: `delta`, `diff`.
+> More information: <https://difftastic.wilfred.me.uk/introduction.html>.
+
+- Compare two files or directories:
+
+`difft {{path/to/original_file_or_directory}} {{path/to/new_file_or_directory}}`
```
Comment (vitorhcl) [inline pages/common/difft.md:9]:
```suggestion
`difft {{path/to/source_file_or_directory}} {{path/to/target_file_or_directory}}`
```

File: pages/common/difft.md [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+
+- Only report the presence of differences between the files:
+
+`difft --check-only {{path/to/original_file}} {{path/to/new_file}}`
```
Comment (vitorhcl) [inline pages/common/difft.md:13]:
```suggestion
`difft --check-only {{path/to/source_file}} {{path/to/target_file}}`
```

File: pages/common/difft.md [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+
+- Specify the display mode (default is `side-by-side`):
+
+`difft --display {{side-by-side|side-by-side-show-both|inline|json}} {{path/to/original_file}} {{path/to/new_file}}`
```
Comment (vitorhcl) [inline pages/common/difft.md:17]:
```suggestion
`difft --display {{side-by-side|side-by-side-show-both|inline|json}} {{path/to/source_file}} {{path/to/target_file}}`
```

File: pages/common/difft.md [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+
+- Ignore comments when comparing:
+
+`difft --ignore-comments {{path/to/original_file}} {{path/to/new_file}}`
```
Comment (vitorhcl) [inline pages/common/difft.md:21]:
```suggestion
`difft --ignore-comments {{path/to/source_file}} {{path/to/target_file}}`
```

File: pages/common/difft.md [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+
+- Enable/Disable syntax highlighting of source code (default is `on`):
+
+`difft --syntax-highlight {{on|off}} {{path/to/original_file}} {{path/to/new_file}}`
```
Comment (vitorhcl) [inline pages/common/difft.md:25]:
```suggestion
`difft --syntax-highlight {{on|off}} {{path/to/source_file}} {{path/to/target_file}}`
```

File: pages/common/difft.md [hunk 12]
```diff
@@ -0,0 +1,33 @@
...
+
+- Do not output anything at all if there are no differences between files:
+
+`difft --skip-unchanged {{path/to/original_file_or_directory}} {{path/to/new_file_or_directory}}`
```
Comment (vitorhcl) [inline pages/common/difft.md:29]:
```suggestion
`difft --skip-unchanged {{path/to/source_file_or_directory}} {{path/to/target_file_or_directory}}`
```

File: pages/common/difft.md [hunk 13]
```diff
@@ -0,0 +1,33 @@
...
+
+- Print all the languages supported by the tool, along with their extensions:
```
Comment (vitorhcl) [inline pages/common/difft.md:31]:
```suggestion
- Print all programmjng languages supported by the tool, along with their extensions:
```

At a first glance, "language" alone can be confused with "idiom"

**Review by vitorhcl — APPROVED** [review]
LGTM after we replace the terms "original" and "new".

I'm open for suggestions of better words than "source" and "target".

File: pages/common/difft.md (hunk 7 above)
Comment (fazlearefin) [reply pages/common/difft.md:9]: Source and target makes sense when transferring or copying files from one location to another. For diff, I don't think it is ideal.

File: pages/common/difft.md (hunk 13 above)
Comment (fazlearefin) [inline pages/common/difft.md:31]:
```suggestion
- Print all programming languages supported by the tool, along with their extensions:
```

Comment (fazlearefin) [reply pages/common/difft.md]: Typo, fixed in change below applying your suggestion.

File: pages/common/difft.md (hunk 7 above)
Comment (fazlearefin) [reply pages/common/difft.md]: Done, used `path/to/file_or_directory1` and `path/to/file_or_directory2`

**Review by kbdharun — APPROVED** [review]

**Review by vitorhcl — APPROVED** [review]

**Pull request merged**